src/Illuminate/Database/Connectors/PostgresConnector.php

```diff
@@ -81,34 +81,42 @@ protected function configureTimezone($connection, array $config)
     }
 
     /**
-     * Set the schema on the connection.
+     * Set the search_path on the connection.
      *
      * @param  \PDO  $connection
      * @param  array  $config
      * @return void
      */
-    protected function configureSchema($connection, $config)
+    protected function configureSearchPath($connection, $config)
     {
-        if (isset($config['schema'])) {
-            $schema = $this->formatSchema($config['schema']);
+        if (isset($config['searchpath']) || isset($config['schema'])) {
+            $searchPath = $this->formatSearchPath($config['searchpath'] ?? $config['schema']);
 
-            $connection->prepare("set search_path to {$schema}")->execute();
+            $connection->prepare("set search_path to {$searchPath}")->execute();
         }
     }
 
     /**
-     * Format the schema for the DSN.
+     * Format the search path for the DSN.
      *
-     * @param  array|string  $schema
+     * @param  array|string  $searchPath
      * @return string
      */
-    protected function formatSchema($schema)
+    protected function formatSearchPath($searchPath)
     {
-        if (is_array($schema)) {
-            return '"'.implode('", "', $schema).'"';
+        if (is_array($searchPath)) {
+            $schemas = array_map(function ($item) {
+                return '"'.$item.'"';
+            }, $searchPath);
+
+            $preparedSchemas = array_map(function ($item) {
+                return ":$item";
+            }, $schemas);
+
+            return implode(', ', array_combine($preparedSchemas, $schemas));
```

Comment:
I don't understand your use of the key=>value approach together with `implode`.

In my book, the keys when performing implode are ignored?

```php
 $ php -v
PHP 7.1.11-1…
…
$ php -r 'var_dump(implode(",", ["a"=>"b", "c" => "d"]));'
string(3) "b,d"
```

I understand you try to use something with prepared statements here but IMHO it ends up being not used.

Reply:
Again, you are spot-on!